lwcomponent/catalog.go

```diff
@@ -305,37 +363,71 @@ func NewCatalog(
 				nil,
 				localHost)
 		} else {
-			// @jon-stewart: persisted API info
-			cdkComponents[localHost.Name()] = NewCDKComponent(localHost.Name(), "", BinaryType, nil, localHost)
+			// Unknown components
+			cdkComponents[localHost.Name()] = NewCDKComponent(localHost.Name(), "", EmptyType, nil, localHost)
 		}
 	}
 
 	return &Catalog{client, cdkComponents, stageConstructor}, nil
 }
 
 func LocalComponents() (components []CDKComponent, err error) {
-	var localHost map[string]HostInfo
+	cdkCache, err := loadCdkCache()
+	if err != nil {
+		return
+	}
+
+	localComponents, err := loadLocalComponents()
+	if err != nil {
+		return
+	}
 
-	localHost, err = loadLocalComponents()
+	// Load components stored in cdk_cache
+	for _, c := range cdkCache {
+		hostInfo, ok := localComponents[c.Name]
+		if ok {
+			delete(localComponents, c.Name)
+		}
+		components = append(components, NewCDKComponent(c.Name, c.Description, c.Type, c.ApiInfo, hostInfo))
+	}
 
-	for _, l := range localHost {
-		if l.Development() {
-			devInfo, err := NewDevInfo(l.Dir())
+	// Load components in the components folder but not stored in cdk_cache
+	for _, hostInfo := range localComponents {
+		if hostInfo.Development() {
+			devInfo, err := NewDevInfo(hostInfo.Dir)
 			if err != nil {
 				return nil, err
 			}
-
-			components = append(components, NewCDKComponent(l.Name(), devInfo.Desc, devInfo.ComponentType, nil, l))
+			components = append(components, NewCDKComponent(hostInfo.Name(), devInfo.Desc, devInfo.ComponentType, nil, hostInfo))
 		} else {
-			// @jon-stewart: persisted API info
-			components = append(components, NewCDKComponent(l.Name(), "", BinaryType, nil, l))
+			// Unknown components
+			components = append(components, NewCDKComponent(hostInfo.Name(), "", EmptyType, nil, hostInfo))
 		}
 	}
 
 	return
 }
 
-func loadLocalComponents() (local map[string]HostInfo, err error) {
+func loadCdkCache() (map[string]CDKComponent, error) {
```

Comment:
Need to handle cache invalidation - when does it expire and when is it replaced.